pkg/performanceprofile/profilecreator/profilecreator.go

```diff
@@ -289,30 +289,108 @@ func topologyHTDisabled(info *topology.Info) *topology.Info {
 	return disabledHTTopology
 }
 
-// GetReservedAndIsolatedCPUs returns Reserved and Isolated CPUs
-func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, splitReservedCPUsAcrossNUMA bool, disableHTFlag bool) (cpuset.CPUSet, cpuset.CPUSet, error) {
+// Calculates the resevered, isolated and offlined cpuSets.
+func (ghwHandler GHWHandler) CalculateCPUSets(reservedCPUCount int, offlinedCPUCount int, splitReservedCPUsAcrossNUMA bool, disableHTFlag bool) (cpuset.CPUSet, cpuset.CPUSet, cpuset.CPUSet, error) {
+
 	cpuInfo, err := ghwHandler.CPU()
 	if err != nil {
-		return cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't obtain CPU info from GHW snapshot: %v", err)
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("can't obtain CPU info from GHW snapshot: %v", err)
 	}
 
+	// Check limits are in range
 	if reservedCPUCount <= 0 || reservedCPUCount >= int(cpuInfo.TotalThreads) {
-		return cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("please specify the reserved CPU count in the range [1,%d]", cpuInfo.TotalThreads-1)
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("please specify the reserved CPU count in the range [1,%d]", cpuInfo.TotalThreads-1)
+	}
+
+	if offlinedCPUCount >= int(cpuInfo.TotalThreads) {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("please specify the offlined CPU count in the range [0,%d]", cpuInfo.TotalThreads-1)
 	}
+
+	if reservedCPUCount+offlinedCPUCount >= int(cpuInfo.TotalThreads) {
+		return cpuset.CPUSet{}, cpuset.CPUSet{}, cpuset.CPUSet{}, fmt.Errorf("please specify the reserved plus offlined CPU count in the range [1,%d]", cpuInfo.TotalThreads-1)
```

Comment:
shouldn't this range be `[0, len(cpuInfo.TotalThreads)-reservedCPUCount-1]` ?

Reply:
yeah you are right

Reply:
well, the offline amount is anyway bound between 1 and (total - reserved), so it seems nicer to let this limitation be visible in the error message that, unless I'm missing something, can otherwise be misleading.
And yes, we totally need to have at least 1 reserved core